frontend/src/components/shared/StreamlitMarkdown/StreamlitMarkdown.tsx

```diff
@@ -271,4 +303,45 @@ export function LinkWithTargetBlank(props: LinkProps): ReactElement {
   )
 }
 
+export function Heading(props: HeadingProtoProps): ReactElement {
+  const { width } = props
+  const { tag, anchor, body } = props.element
+  const isSidebar = React.useContext(IsSidebarContext)
+  // st.header can contain new lines which are just interpreted as new
+  // markdown to be rendered as such.
+  const [heading, ...rest] = body.split("\n")
+
+  return (
+    <div className="stMarkdown" style={{ width }}>
+      <HeadingWithAnchor tag={tag} anchor={anchor}>
+        <RenderedMarkdown
+          source={heading}
+          allowHTML={false}
+          // this is purely an inline string
```

Comment:
overrideComponents will take html components OR markdown (because markdown is converted to html) and then will render foo component where foo would be CustomHeading, Fragment, etc for the specified tag (h1, h2, p, span, etc). 

If people add a markdown image, it will work the same as rendered markdown, which you can check by pasting this code 

`![Minion](https://octodex.github.com/images/minion.png)`

to this app: 

https://streamlit-core-previews-streamlit-app-fix4701-preview-sk4lwy.streamlitapp.com/

However, table will not work. Though, a table in a header sounds like a bizarre use case. They *should* be using markdown.